### ISSUE TITLE
cntk is failing while training model with huge data

### ISSUE THREAD

**Issue by datavizweb** [Microsoft/CNTK#924]
Hi,

I am training an acoustic model for speech recognition. I am using Kaldi reader and can train the model (`DNN model, 429:4*1024*4021`) using 20 hours of data. However, when I increase the data to 1000 hours, it consumes all the system RAM (not GPU RAM) and gets killed by the system. It is getting killed while reading the features and alignment (first step). What is the best way to train DNN model for larger data set. I however, not tried with HTKMLF format reader.

Following is the system configuration:

`2 GTX GPU with 4 GB RAM each, 64 GB RAM, 30 CPU cores`. I am running two parallel process with mpiexec (mpiexec -np cntk config=CNTK.cntk). Also I tried with just one process and even it fails to start the model training. I am using minibatch of 4096 (tried also with smaller minibatch size).

What is the best way to train the model with huge amount of data ?. 

`PS: I can use Kaldi tool to train AM using all the 1000 hours data on same system.`

**Comment by n17s** [comment]
The way to do it is to use the [randomize and randomizationWindow options](https://github.com/Microsoft/CNTK/wiki/Understanding-and-Extending-Readers#configuration-options)

**Comment by datavizweb** [comment]
I did experiment with the randomize and randomizationWindow however it didn't work. There are some other issues with KaldiReader (May be I am not using it as required). I am now using the HTKMLF reader and there are no issues with it. I will experiment with the KaldiReader some times later. In any case, thanks for your reply.

Side Note : It is mentioned in the kaldi reader that only scp work for feature specifier. However, I tried with the ark (combined with add-delta, compute-cmvn) both of them work.
